### DISCUSSION TITLE
Workflow QOL Suggestion

### DISCUSSION THREAD

**Discussion by markbland82** [lllyasviel/stable-diffusion-webui-forge#2353] [Ideas]
Just wondering if it would be possible to add QOL features that make it easier to pick up where you left off on an image/workflow. Ie, keep the image pathing info etc for controlnets and img2img. This way when loading an image in to pick up where I left off, I can drop it into PNG info, and push the data to img2img or txt2img etc and the corresponding images that were in that workflow will populate along with their respective settings. It becomes pretty impossible to keep track of a workflow (ie what images were in controlnet/ipadapter and/or img2img) months down the line imo. Would be an amazing feature and one that would help this tool become more robust reliable from a workflow standpoint. 

Thanks

**Comment by Profileiche** [comment]
What you could be trying is the "State Manager" (https://github.com/SenshiSentou/sd-webui-state-manager) extension. It is currently a little buggy, but works like what you asked for.

Another alternative is: https://github.com/ilian6806/stable-diffusion-webui-state

You might also want to press the little blue button beneath the Generate button, as it recalls the last generation settings, but alas not all Control Net setups.

**Reply by markbland82** [reply]
Thanks, but realistically all that needs to happen is the images are saved with the pathing for the various img2img/controlnets images used. Currently I get all the settings recalled from the png info for basically everything, except the images that were associated to them. If the image has been moved/renamed it can show a broken URL with the path/name of the file, but otherwise it would be great to see the relevant images load into the workflow so I can continue where I left off.

**Reply by Profileiche** [reply]
That would certainly be a QoL change. I wonder if there is a reason it isn't implemented yet.
